sdk/freestanding-execution-engine/src/main.rs

```diff
@@ -124,49 +121,44 @@ fn parse_command_line() -> Result<CommandLineOptions, Box<dyn Error>> {
                 .help(
                     "Selects the execution strategy to use: interpretation or JIT (defaults to \
                      interpretation).",
-                )
-                .required(false)
-                .multiple(false)
-                .default_value("jit"),
+                ),
         )
         .arg(
             Arg::with_name("dump-stdout")
                 .short("o")
                 .long("dump-stdout")
                 .help("Whether the contents of stdout should be dumped before exiting")
-                .required(true)
-                .value_name("BOOLEAN")
-                .default_value(&default_dump_stdout),
+                .value_name("BOOLEAN"),
         )
         .arg(
             Arg::with_name("dump-stderr")
                 .short("e")
                 .long("dump-stderr")
                 .help("Whether the contents of stderr should be dumped before exiting")
-                .required(true)
-                .value_name("BOOLEAN")
-                .default_value(&default_dump_stderr),
+                .value_name("BOOLEAN"),
         )
         .get_matches();
 
     info!("Parsed command line.");
 
-    let execution_strategy = if let Some(strategy) = matches.value_of("execution-strategy") {
-        if strategy == "interp" {
-            info!("Selecting interpretation as the execution strategy.");
-            ExecutionStrategy::Interpretation
-        } else if strategy == "jit" {
-            info!("Selecting JITting as the execution strategy.");
-            ExecutionStrategy::JIT
-        } else {
-            return Err(format!(
-                "Expecting 'interp' or 'jit' as selectable execution strategies, but found {}",
-                strategy
-            )
-            .into());
+    let execution_strategy = {
+        let strategy = matches.value_of("execution-strategy").unwrap_or("jit");
+        match strategy {
+            "interp" => {
+                info!("Selecting interpretation as the execution strategy.");
+                ExecutionStrategy::Interpretation
+            }
+            "jit" => {
+                info!("Selecting JITting as the execution strategy.");
+                ExecutionStrategy::JIT
+            }
+            _ => {
```

Comment:
Isn't this case now impossible, due to `strategy` defaulting to `jit` in the `unwrap_or()`?  If so, I would explicitly mark this as impossible with `unreachable!()`.

Reply:
It can happen if the user specifies something else:
```
$ cargo run -- --program read-file.wasm -x foobar
    Finished dev [unoptimized + debuginfo] target(s) in 0.58s
     Running `target/debug/freestanding-execution-engine --program read-file.wasm -x foobar`
Error: "Expecting \'interp\' or \'jit\' as selectable execution strategies, but found foobar"
```

(There are too many quotes in the error message, really, but all the error messages are generated in the same way.)

Reply:
Ah OK, yeah fair enough!